src/modules/fundingManager/bondingCurveFundingManager/BondingCurveFundingManagerBase.sol

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.19;
+
+// Internal Dependencies
+import {Module} from "src/modules/base/Module.sol";
+import {IBondingCurveFundingManagerBase} from
+    "src/modules/fundingManager/bondingCurveFundingManager/IBondingCurveFundingManagerBase.sol";
+
+// Internal Interfaces
+import {IOrchestrator} from "src/orchestrator/IOrchestrator.sol";
+import {ContextUpgradeable} from "@oz-up/utils/ContextUpgradeable.sol";
+
+// External Interfaces
+import {IERC20} from "@oz/token/ERC20/IERC20.sol";
+import {IERC20MetadataUpgradeable} from
+    "@oz-up/token/ERC20/extensions/IERC20MetadataUpgradeable.sol";
+
+// External Dependencies
+import {ERC20Upgradeable} from "@oz-up/token/ERC20/ERC20Upgradeable.sol";
+
+// External Libraries
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
+import {Strings} from "@oz/utils/Strings.sol";
+
+import {IFundingManager} from "src/modules/fundingManager/IFundingManager.sol";
+
+abstract contract BondingCurveFundingManagerBase is
+    IBondingCurveFundingManagerBase,
+    IFundingManager,
+    ContextUpgradeable,
+    ERC20Upgradeable,
+    Module
+{
+    using Strings for uint;
+    using SafeERC20 for IERC20;
+
+    //--------------------------------------------------------------------------
+    // Storage
+    uint8 internal tokenDecimals;
+    bool internal buyIsOpen;
+    uint internal buyFee;
+    address internal collateral;
+    // Q: Do we want to set other beneficiary in the contract then the module itself for the fees?
```

Comment:
I think transferOrchestratorToken() is good enough to move the fees out when the time comes (if the fees are going to go to the worklflow owners anyway). We'd probably need a way to track the collected amount, though

Reply:
I agree and will not be implementing a beneficiary on top of the standard way to pull out tokens